### PR TITLE
Replace forked cache by official npm

### LINKED ISSUE #14: Use npm package actions/cache
Instead of the forked one, see actions/cache#293

### PR DESCRIPTION
**Pull request by sayboras**
Fixes #14 

## Testing

https://github.com/sayboras/golangci-lint-action/runs/698981324?check_suite_focus=true

<details>
<summary>Restore cache</summary>

```
prepare environment
  Finding needed golangci-lint version...
  Setup go stable version spec 1
  A version satisfying 1 not found locally, attempting to download ...
  Requested golangci-lint 'v1.26', using 'v1.26.0', calculation took 286ms
  Installing golangci-lint v1.26.0...
  Downloading https://github.com/golangci/golangci-lint/releases/download/v1.26.0/golangci-lint-1.26.0-linux-amd64.tar.gz ...
  Downloading from https://storage.googleapis.com/golang/go1.14.3.linux-amd64.tar.gz
  Cache Size: ~0 MB (28191 B)
  /bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/55990ecc-a52b-42fb-928d-26f56fe0343f/cache.tzst -P -C /home/runner/work/golangci-lint-action/golangci-lint-action
  Writing patch to /tmp/tmp-2722-n5jtq19CBJj1/pull.patch
  Restored cache for golangci-lint from key 'golangci-lint.cache-2629-c44fb3d8f525e6bb1b453fa82b0f85dd991994c8' in 390ms
  /bin/tar xz --warning=no-unknown-keyword -C /home/runner -f /home/runner/work/_temp/99660000-eead-464d-b15f-53111516907c
  Installed golangci-lint into /home/runner/golangci-lint-1.26.0-linux-amd64/golangci-lint in 689ms
  Extracting ...
  /bin/tar xz --warning=no-unknown-keyword -C /home/runner/work/_temp/8b9176f0-4db7-494e-8d86-85de3c1c65fd -f /home/runner/work/_temp/666ada13-4b87-4f67-87c9-e3a6372c4e60
```

</details>

<details>
<summary>Save cache</summary>

```
Post job cleanup.
/bin/tar --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/golangci-lint-action/golangci-lint-action --files-from manifest.txt
Cache saved successfully
Saved cache for golangci-lint from paths '/home/runner/.cache/golangci-lint, /home/runner/.cache/go-build, /home/runner/go/pkg' in 410ms

....
Post job cleanup.
[warning] reserveCache failed: Cache already exists. Scope: refs/pull/1/merge, Key: golangci-lint.cache-2629-c44fb3d8f525e6bb1b453fa82b0f85dd991994c8, Version: 0f79570b790d0689c186c2d92be94cb909971c6643d398173c3c5a0ecd5cf601
```

</details>

### CODE REVIEW

**Comment by jirfag** [comment]
hi, thank you!

shouldn't we implement the same logic as in [@actions/cache](https://github.com/actions/cache). E.g. I see that we didn't implement [this check](https://github.com/actions/cache/blob/master/src/save.ts#L27) and it lead to `[warning] reserveCache failed: Cache already exists. Scope: refs/pull/1/merge, Key: golangci-lint.cache-2629-c44fb3d8f525e6bb1b453fa82b0f85dd991994c8, Version: 0f79570b790d0689c186c2d92be94cb909971c6643d398173c3c5a0ecd5cf601`

**Comment by sayboras** [comment]
> hi, thank you!
> 
> shouldn't we implement the same logic as in [@actions/cache](https://github.com/actions/cache)? E.g. I see that we didn't implement [this check](https://github.com/actions/cache/blob/master/src/save.ts#L27) and it lead to `[warning] reserveCache failed: Cache already exists. Scope: refs/pull/1/merge, Key: golangci-lint.cache-2629-c44fb3d8f525e6bb1b453fa82b0f85dd991994c8, Version: 0f79570b790d0689c186c2d92be94cb909971c6643d398173c3c5a0ecd5cf601`

Oh right, let me make an update.

**Comment by sayboras** [comment]
I have created one dummy PR for verifying again, please find below details:

- first run https://github.com/sayboras/golangci-lint-action/runs/699826911?check_suite_focus=true
- subsequent run https://github.com/sayboras/golangci-lint-action/actions/runs/112520402

File: src/cache.ts [hunk 1]
```diff
@@ -52,30 +54,82 @@ async function buildCacheKeys(): Promise<string[]> {
...
+  try {
+    const cacheKey = await cache.restoreCache(getCacheDirs(), primaryKey, restoreKeys)
+    if (!cacheKey) {
+      core.info(`Cache not found for input keys: ${[primaryKey, ...restoreKeys].join(", ")}`)
+      return
+    }
+    // Store the matched cache key
+    utils.setCacheState(cacheKey)
+    const isExactKeyMatch = utils.isExactKeyMatch(primaryKey, cacheKey)
+    utils.setCacheHitOutput(isExactKeyMatch)
```
Comment (jirfag) [inline src/cache.ts:87]: please, remove it: I don't want this to be a part of an exported interface of our GitHub action

**Comment by jirfag** [comment]
thank you very much!

**Pull request merged**